File: src/main/java/io/github/chaosdave34/kitpvp/commands/CustomItemTabCompleter.java
package io.github.chaosdave34.kitpvp.commands;

import io.github.chaosdave34.kitpvp.KitPvp;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.command.TabCompleter;
import org.bukkit.util.StringUtil;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class CustomItemTabCompleter implements TabCompleter {
    @Override
    public @Nullable List<String> onTabComplete(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if (args.length == 1) {
            List<String> completions = new ArrayList<>();
            StringUtil.copyPartialMatches(args[0], KitPvp.INSTANCE.getCustomItemHandler().getCustomItems().keySet(), completions);
            return completions;
        }

        return Collections.emptyList();
    }

}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/commands/CustomItemTabCompleter.kt
package io.github.chaosdave34.kitpvp.commands

import io.github.chaosdave34.kitpvp.KitPvp
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.command.TabCompleter
import org.bukkit.util.StringUtil

class CustomItemTabCompleter : TabCompleter {
    override fun onTabComplete(sender: CommandSender, command: Command, label: String, args: Array<String>): List<String> {
        val completions: MutableList<String> = mutableListOf()
        if (args.size == 1) {
            StringUtil.copyPartialMatches(args[0], KitPvp.INSTANCE.customItemHandler.customItems.keys, completions)
            return completions
        }
        return completions
    }
}